bigquery/snippets/managedwriter/bigquerystorage_append_rows_pending.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package managedwriter
+
+// [START bigquerystorage_append_rows_pending]
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math/rand"
+	"time"
+
+	"cloud.google.com/go/bigquery/storage/managedwriter"
+	"cloud.google.com/go/bigquery/storage/managedwriter/adapt"
+	"github.com/GoogleCloudPlatform/golang-samples/bigquery/snippets/managedwriter/exampleproto"
+	storagepb "google.golang.org/genproto/googleapis/cloud/bigquery/storage/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+// generateExampleMessage generates an example protobuf message using a statically defined and
+// compiled protocol buffer file, and returns the binary serialized representation.
+func generateExampleMessage() ([]byte, error) {
+
+	random := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	// Our example data embeds an array of structs, so generate that first.
+	sList := make([]*exampleproto.SampleData_SampleStruct, 5)
+	for i := 0; i < 5; i++ {
+		sList[i] = &exampleproto.SampleData_SampleStruct{
+			SubIntCol: proto.Int64(random.Int63()),
+		}
+	}
+
+	m := &exampleproto.SampleData{
+		BoolCol:    proto.Bool(true),
+		BytesCol:   []byte("some bytes"),
+		Float64Col: proto.Float64(3.14),
+		Int64Col:   proto.Int64(123),
+		StringCol:  proto.String("example string value"),
+
+		// These types require special encoding/formatting to transmit.
+
+		// DATE values are number of days since the Unix epoch.
+		DateCol: proto.Int32(int32(time.Now().UnixMilli() / 86400000)),
+
+		// DATETIME uses the literal format.
+		DatetimeCol: proto.String("2022-01-01 12:13:14.000000"),
+
+		// GEOGRAPHY uses Well-Known-Text (WKT) format.
+		GeographyCol: proto.String("POINT(-122.350220 47.649154)"),
+
+		// NUMERIC and BIGNUMERIC can be passed as string, or more efficiently
+		// using a packed byte representation.
+		NumericCol:    proto.String("99999999999999999999999999999.999999999"),
```

Comment:
We intentionally used a rather small schema in `bigquerystorage_append_rows_pending`. 

https://github.com/googleapis/python-bigquery-storage/blob/main/samples/snippets/append_rows_pending.py

`bigquerystorage_append_rows_raw_proto2` has a more thorough schema

https://github.com/googleapis/python-bigquery-storage/blob/main/samples/snippets/append_rows_proto2.py

but indeed perhaps a different name is desired for Go since it's less "raw" than Python.

Reply:
Switched to region tag bigquerystorage_write_pending_complexschema